desktop-src/direct3dhlsl_current/hlsl-method-gatherCmp.md

```diff
@@ -0,0 +1,283 @@
+   [, out uint          status ] );
+
+ElementType Texture2DArray<ElementType>::GatherCmpAlpha(
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+      in  int2          offset,
+   [, in  int2          offset2]
+   [, in  int2          offset3]
+   [, in  int2          offset4]
+   [, out uint          status ] );
+```
+
+#### TextureCube
+```syntax
+// same as GatherRed
+ElementType TextureCube<ElementType>::GatherCmp(
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCube<ElementType>::GatherCmpRed(
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCube<ElementType>::GatherCmpGreen(
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCube<ElementType>::GatherCmpBlue(
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCube<ElementType>::GatherCmpAlpha(
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+```
+
+#### TextureCubeArray
+```syntax
+// same as GatherRed
+ElementType TextureCubeArray<ElementType>::GatherCmp(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpRed(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpGreen(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpBlue(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpAlpha(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+```
+
+## Return Value
+
+The texture's template type for the element, which may be a scalar or vector.
+This provides the HLSL type for an element loaded from a texture.
+The format stored in the resource is based on the texture resource view's DXGI_FORMAT,
+which will determine how data is translated into the type specified in HLSL.
+
+## Parameters
+
+
+| Parameter | Description |
+| - | - |
+| in [`SamplerState samp`](#samplerstate-samp) | SamplerState object that determines how sampling will take place. |
```

Comment:
This should be `SamplerComparisonState` for `GatherCmp*`.

Reply:
Thank you. Done.